clientapi/auth/storage/accounts/account_data_table.go

```diff
@@ -120,28 +120,20 @@ func (s *accountDataStatements) selectAccountData(
 
 func (s *accountDataStatements) selectAccountDataByType(
 	ctx context.Context, localpart, roomID, dataType string,
-) (data []gomatrixserverlib.ClientEvent, err error) {
-	data = []gomatrixserverlib.ClientEvent{}
-
+) (data *gomatrixserverlib.ClientEvent, err error) {
 	stmt := s.selectAccountDataByTypeStmt
-	rows, err := stmt.QueryContext(ctx, localpart, roomID, dataType)
+	row := stmt.QueryRowContext(ctx, localpart, roomID, dataType)
```

Comment:
```suggestion
```

This line and the `Scan` statement below can be merged into the `if` block.

Reply:
indeed it would be much better, i will change that.